test/test.js

```diff
@@ -1044,11 +1044,25 @@ describe('Page', function() {
       await waitFor;
       expect(found).toBe(true);
     });
-    it('should not allow you to select an element with single slash xpath', async({page, server}) => {
+    it('should wait for an absolute xpath', async({page, server}) => {
+      let found = false;
       await page.setContent(`<div>some text</div>`);
-      let error = null;
-      await page.waitFor('/html/body/div').catch(e => error = e);
-      expect(error).toBeTruthy();
+      await page.waitFor('/html/body/div').then(() => found = true);
```

Comment:
XPath.

But are those CSS comments something we wish to support in determent of plain XPaths?

Reply:
We could have a negative lookahead in the pattern: `(?!^[/][*])^[(]?(?:[.][/]|[/][/]?|id[(])`:

- https://regex101.com/r/OQXuZ9/3

But the problem then would be that XPaths like `/*/div` would not be considered.

Reply:
> But are those CSS comments something we wish to support in determent of plain XPaths?

It would be nice to have a good xpath support. However, CSS comments don't look like an exotic scenario to me, so we shouldn't break it until the next major release.

Reply:
IMO, I don't know why anyone would wait for a CSS selector with a comment, but YMMV. 

Still, do you think we could introduce the negative lookahead to deal with these comments?
It would still be an improvement XPath-wise and it wouldn't break the comment scenario.

Reply:
@alixaxel I don't think the negative lookahead will cover all the usecases.

Let's not complicate the heuristics here for now. We can revisit it later if there's a demand.